rclcpp/src/rclcpp/publisher_base.cpp

```diff
@@ -384,3 +384,22 @@ std::vector<rclcpp::NetworkFlowEndpoint> PublisherBase::get_network_flow_endpoin
 
   return network_flow_endpoint_vector;
 }
+
+size_t PublisherBase::lowest_available_ipm_capacity() const
+{
+  if (!intra_process_is_enabled_) {
+    return std::numeric_limits<size_t>::max();
```

Comment:
should this be `0` if intra-process is disabled?

Reply:
the idea was if intraprocess is not enabled, the capacity should be high to avoid blocking (in the case you are waiting for free buffers as they will never become free)